02-peer-protocol.md

```diff
@@ -238,6 +238,18 @@ would be eliminated as dust.  The similar requirements in
 `accept_channel` ensure that both sides' `channel_reserve_satoshis`
 are above both `dust_limit_satoshis`.
 
+#### Practical Considerations for temporary_channel_id
+
+Note that as duplicate `temporary_channel_id`s may exist from different
+peers, APIs which reference channels by their channel id before the funding
+transaction is created are inherently unsafe. The only protocol-provided
+identifier for a channel before funding_created has been exchanged is the
+(source_node_id, destination_node_id, temporary_channel_id) tuple. Note that
+any such APIs which reference channels by their channel id before the funding
+transaction is confirmed are also not persistent - until you know the script
+pubkey corresponding to the funding txo nothing prevents duplicative channel
+ids.
```

Comment:
wouldn't it make sense to add a statement that peers MUST NOT share the `temporary_channel_id` with other peers and that peers MUST ignore `temporary_channel_id`s if learnt about from other peers?

Reply:
This isn't true; you can share these all you want.  You can hard-code them in the source if you only allow one channel open at once.  If they were required to be globally unique, we'd have to make strong conditions around them...